packages/shared-react/src/api/apiUtils.ts

```diff
@@ -39,3 +48,85 @@ export const getFulfilledValues = <T>(results: PromiseSettledResult<T>[]) =>
   results
     .filter((result): result is PromiseFulfilledResult<T> => result.status === 'fulfilled')
     .map(({ value }) => value)
+
+export const getSignerMissingBalances = async (
+  expectedBalances: Map<string, bigint>,
+  signerAddress: string,
+  networkId: number | undefined
+) => {
+  const missingBalances: Map<string, bigint> = new Map()
+
+  const { balances: signerBalances } = await queryClient.fetchQuery(
+    addressBalancesQuery({ addressHash: signerAddress, networkId })
+  )
+
+  for (const [tokenId, amount] of expectedBalances) {
+    const balance = signerBalances.find((balance) => balance.id === tokenId)
+
+    if (!balance) {
+      missingBalances.set(tokenId, amount)
+    } else if (BigInt(balance.availableBalance) < amount) {
+      missingBalances.set(tokenId, amount - BigInt(balance.availableBalance))
+    }
+  }
+
+  return missingBalances
+}
+
+export const getAddressWithEnoughBalances = async (
+  missingBalances: Map<string, bigint>,
```

Comment:
Nit: to make it easier to understand what's in the map, we could use `expectedBalances: Map<Token['id'], bigint>`.

Reply:
done